### PR TITLE
Drop data points with start and end time too close together

### PR DESCRIPTION
**Pull request by dashpole**
See breaking-changes.md for the rationale for this change.

### CODE REVIEW

File: exporter/collector/metricsexporter.go [hunk 1]
```diff
@@ -350,7 +350,11 @@ func (m *metricMapper) metricToTimeSeries(
 		sum := metric.Sum()
 		points := sum.DataPoints()
 		for i := 0; i < points.Len(); i++ {
-			ts := m.sumPointToTimeSeries(resource, extraLabels, metric, sum, points.At(i))
+			point := points.At(i)
+			if !intervalIsValid(point.StartTimestamp().AsTime(), point.Timestamp().AsTime()) {
+				continue
+			}
```
Comment (aabmass) [inline exporter/collector/metricsexporter.go:356]: Would it make more sense to move this logic into the corresponding `xPointToTimeSeries()` methods?

File: exporter/collector/breaking-changes.md [hunk 2]
```diff
@@ -50,6 +50,16 @@ For now, it is not possible to customizate the mapping algorithm, beyond using t
...
+
+```When StartTimeUnixNano equals TimeUnixNano, a new unbroken sequence of observations begins with a reset at an unknown start time. The initial observed value is recorded to indicate that an unbroken sequence of observations resumes. These points have zero duration, and indicate that nothing is known about previously-reported points and that data may have been lost.```
+
+The OpenCensus-based exporter took points with identical start and end times, and added a millisecond to the end time: https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/pull/287.  However, the resulting point makes little sense: It has a very small time period, and a normal cumulative value for an interval.  The result is that graphs display a very large rate over that millisecond, which is misleading.
+
+This exporter instead drops points with identical start and end times.  Resets are already indicated by a change in the start timestamp for the subsequent point.
```
[the remark this answers is not in the document]
Comment (dashpole) [reply exporter/collector/breaking-changes.md:61]: I will do more experimentation to ensure this does what we want, or propose an alternative.

File: exporter/collector/metricsexporter.go (hunk 1 above)
Comment (dashpole) [reply exporter/collector/metricsexporter.go:356]: I thought about that, but then we would need to check the return value of xPointToTimeSeries() anyways.

Comment (aabmass) [reply exporter/collector/metricsexporter.go:356]: Ya, another option is having them return `[]*TimeSeries` so `append()` will do the right thing. Up to you, just calling it out

File: exporter/collector/breaking-changes.md [hunk 3]
```diff
@@ -50,6 +50,16 @@ For now, it is not possible to customizate the mapping algorithm, beyond using t
 in the collector pipeline before the exporter. If you have a use case for customizing the
 behavior, please open an issue.
 
+## Time staggering
+
+The [OTel datamodel specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#resets-and-gaps) an equal start and end time on a cumulative metric indicate a reset timeseries:
```
Comment (punya) [inline exporter/collector/breaking-changes.md:55]:
```suggestion
According to the [OTel datamodel specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#resets-and-gaps), an equal start and end time on a cumulative metric indicate a reset timeseries:
```

File: exporter/collector/breaking-changes.md [hunk 4]
```diff
@@ -50,6 +50,16 @@ For now, it is not possible to customizate the mapping algorithm, beyond using t
...
+
+```When StartTimeUnixNano equals TimeUnixNano, a new unbroken sequence of observations begins with a reset at an unknown start time. The initial observed value is recorded to indicate that an unbroken sequence of observations resumes. These points have zero duration, and indicate that nothing is known about previously-reported points and that data may have been lost.```
```
Comment (punya) [inline exporter/collector/breaking-changes.md:57]:
```suggestion
> When StartTimeUnixNano equals TimeUnixNano, a new unbroken sequence of observations begins with a reset at an unknown start time. The initial observed value is recorded to indicate that an unbroken sequence of observations resumes. These points have zero duration, and indicate that nothing is known about previously-reported points and that data may have been lost.
```

**Review by punya — APPROVED** [review]

File: exporter/collector/metricsexporter.go (hunk 1 above)
Comment (dashpole) [reply exporter/collector/metricsexporter.go]: done

File: exporter/collector/breaking-changes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsuereth) [reply exporter/collector/breaking-changes.md:61]: `normalize_sums_processor` which takes the first point, remembers its value and timestamp, then just reports cumulative diffs from that point.

I think we may want to add that feature directly into the exporter (we can discuss in chat).

**Pull request closed without merging**